base/array.jl

```diff
@@ -863,22 +863,9 @@ function indexin(a::AbstractArray, b::AbstractArray)
     [get(bdict, i, 0) for i in a]
 end
 
-# findin (the index of intersection)
-function findin(a, b::UnitRange)
-    ind = Array(Int, 0)
-    f = first(b)
-    l = last(b)
-    for i = 1:length(a)
-        if f <= a[i] <= l
-            push!(ind, i)
-        end
-    end
-    ind
-end
-
 function findin(a, b)
     ind = Array(Int, 0)
-    bset = Set(b)
+    bset = unique(b)
     @inbounds for i = 1:length(a)
         a[i] in bset && push!(ind, i)
```

Comment:
These are some good points, and actually makes me want to pursue a slightly different optimization. The existing generic `findin` method is really doing two things in the call to `Set`. It is (1) pulling out a set of unique elements in `b` (this is worst case `O(N log N)`) and (2) putting them into an efficient data structure to test for membership (`in` costs `O(log k)` for k elements int the set). I think this is a pretty good strategy when `b` is a dense or array or similar. 

These (getting the unique elements and testing membership) are separate optimizations. As was mentioned in the initial PR rationale, I view optimization of membership testing as a job for specialized `in` methods dispatching on the type of `b`, and the extant implementation does this. with regard to the unique optimization, one can imagine adding a new function `findinunique(a,b)`, in which the argument `b` is a collection (but not necessarily a `Set`) that has unique elements and does all the work testing for membership. The job of `findin` would then becoming making `b` unique in some optimized way before passing it to `findinunique` (which need not be exported).